plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.php

```diff
@@ -148,48 +146,63 @@ public function createSubmissionFileNode($doc, $submissionFile)
         $this->createLocalizedNodes($doc, $submissionFileNode, 'subject', $submissionFile->getData('subject'));
 
         // If it is a dependent file, add submission_file_ref element
-        if ($submissionFile->getData('fileStage') == SubmissionFile::SUBMISSION_FILE_DEPENDENT && $submissionFile->getData('assocType') == ASSOC_TYPE_SUBMISSION_FILE) {
+        if ($submissionFile->getData('fileStage') == SubmissionFile::SUBMISSION_FILE_DEPENDENT && $submissionFile->getData('assocType') == PKPApplication::ASSOC_TYPE_SUBMISSION_FILE) {
             $fileRefNode = $doc->createElementNS($deployment->getNamespace(), 'submission_file_ref');
             $fileRefNode->setAttribute('id', $submissionFile->getData('assocId'));
             $submissionFileNode->appendChild($fileRefNode);
         }
 
         // Create the revision nodes
         $revisions = Repo::submissionFile()->getRevisions(($submissionFile->getId()));
+        $basePath = rtrim(Config::getVar('files', 'files_dir'), '/') . '/';
+        $hasRevision = false;
         foreach ($revisions as $revision) {
-            $localPath = rtrim(Config::getVar('files', 'files_dir'), '/') . '/' . $revision->path;
+            $localPath = $basePath . $revision->path;
+            if (!file_exists($localPath)) {
```

Comment:
So here we are not skipping the `SubmissionFile` if a revision specific file is missing, we are skipping the revision, right?

Reply:
Right, I'm trying to export as much as possible, so if there's at least one revision, it will be exported. Then we'll just show some warnings to let the user know we did the best we could.

On the forum I saw it's very common to have some missing files, especially from people that migrated from OJS 2.x.
Their journal is healthy and the production files are in place, but they have a bunch of missing files (perhaps orphan data which will be cleaned at OJS 3.4?!).